File: WebSalesLibraries/protected/models/qpages/models/QPageModel.php
<?php

	/**
	 * Class QPageModel
	 */
	class QPageModel
	{
		/**
		 * @var string id
		 * @soap
		 */
		public $id;
		/**
		 * @var string title
		 * @soap
		 */
		public $title;
		/**
		 * @var boolean isEmail
		 * @soap
		 */
		public $isEmail;
		/**
		 * @var string url
		 * @soap
		 */
		public $url;
		/**
		 * @var string createDate
		 * @soap
		 */
		public $createDate;
		/**
		 * @var string expirationDate
		 * @soap
		 */
		public $expirationDate;
		/**
		 * @var string subtitle
		 * @soap
		 */
		public $subtitle;
		/**
		 * @var string header
		 * @soap
		 */
		public $header;
		/**
		 * @var string footer
		 * @soap
		 */
		public $footer;
		/**
		 * @var boolean isRestricted
		 * @soap
		 */
		public $isRestricted;
		/**
		 * @var boolean disableBanners
		 * @soap
		 */
		public $disableBanners;
		/**
		 * @var boolean disableWidgets
		 * @soap
		 */
		public $disableWidgets;
		/**
		 * @var boolean showLinksAsUrl
		 * @soap
		 */
		public $showLinksAsUrl;
		/**
		 * @var boolean recordActivity
		 * @soap
		 */
		public $recordActivity;
		/**
		 * @var string pinCode
		 * @soap
		 */
		public $pinCode;
		/**
		 * @var string activityEmailCopy
		 * @soap
		 */
		public $activityEmailCopy;
		/**
		 * @var string Logo
		 * @soap
		 */
		public $logo;
		/**
		 * @var QPageLinkModel[]
		 * @soap
		 */
		public $links;

		/**
		 * @var string login
		 * @soap
		 */
		public $login;
		/**
		 * @var string firstName
		 * @soap
		 */
		public $firstName;
		/**
		 * @var string lastName
		 * @soap
		 */
		public $lastName;
		/**
		 * @var string email
		 * @soap
		 */
		public $email;
		/**
		 * @var string groups
		 * @soap
		 */
		public $groups;
	}

File: SalesLibraries.ServiceConnector/Models/Soap/QBuilder/QPageModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SalesLibraries.ServiceConnector.QBuilderService
{
	public partial class QPageModel
	{
		public bool FullyLoaded { get; set; }

		public string FullName => (firstName + " " + lastName).Trim();

		public string Type => isEmail ? "Email" : "quickSITE";

		public string SecurityType => isRestricted ? "protected" : "public";

		public string[] GroupNameList
		{
			get
			{
				return !string.IsNullOrEmpty(groups) ? groups.Split(',').Select(x => x.Trim()).ToArray() : new string[] { };
			}
		}

		public DateTime? CreateDate
		{
			get
			{
				DateTime temp;
				if (DateTime.TryParse(createDate, out temp))
					return temp;
				return null;
			}
		}

		public DateTime? ExpirationDate
		{
			get
			{
				DateTime temp;
				if (DateTime.TryParse(expirationDate, out temp))
					return temp;
				return null;
			}
		}

		public int? TotalViews => totalViews > 0 ? (int?)totalViews : null;

		public string Details
		{
			get
			{
				var result = new StringBuilder();
				if (CreateDate.HasValue)
					result.AppendLine(String.Format("{0} - Created: {1}", Type, CreateDate.Value.ToString("MM/dd/yy hh:mm tt")));
				return result.ToString();
			}
		}

		public void RemoveLink(QPageLinkModel link)
		{
			if (link != null)
			{
				var linkList = new List<QPageLinkModel>();
				linkList.AddRange(links);
				linkList.Remove(link);
				links = linkList.ToArray();
			}
		}
	}
}